src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,215 @@
+#include <QObject>
+
+#include "broadcast/scrobblingmanager.h"
+#include "control/controlproxy.h"
+#include "engine/enginexfader.h"
+#include "mixer/deck.h"
+
+ScrobblingManager::ScrobblingManager() :
+m_CPGuiTick("[Master]", "guiTick50ms",this),
+m_CPCrossfader("[Master]","crossfader", this),
+m_CPXFaderCurve(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                "xFaderCurve"),this),
+
+m_CPXFaderCalibration(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                      "xFaderCalibration"),this),
+
+m_CPXFaderMode(ConfigKey(EngineXfader::kXfaderConfigKey, 
+               "xFaderMode"),this),
+
+m_CPXFaderReverse(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                  "xFaderReverse"),this)
+{
+    m_CPGuiTick.connectValueChanged(SLOT(slotGuiTick(double)));
+    startTimer(1000);
+}
+
+
+void ScrobblingManager::slotTrackPaused(TrackPointer pPausedTrack) {
+    //Extra functional decision to only track main decks.            
+    Deck *sourceDeck = qobject_cast<Deck*>(sender());    
+    if (sourceDeck == 0) {
+        qDebug() << "Didn't load track in a deck yet scrobbling "
+                    "received paused signal.";
+        return;
+    }
+    // QMutexLocker locker(&m_mutex);
+    bool allPaused = true;
+    TrackInfo *pausedTrackInfo;
+    for (TrackInfo *trackInfo : m_trackList) {
+        if (trackInfo->m_pTrack == pPausedTrack) {
+            pausedTrackInfo = trackInfo;
+            for (BaseTrackPlayer *player : trackInfo->m_players) {
+                BaseTrackPlayerImpl *playerImpl = 
+                    qobject_cast<BaseTrackPlayerImpl*>(player);
+                if (playerImpl == 0) {
+                    qDebug() << "Track player interface isn't a "
+                                "BaseTrackPlayerImpl";
+                    return;
+                }
+                if (!playerImpl->isTrackPaused())
+                    allPaused = false;
+            }
+            break;
+        }                   
+    }
+    if (allPaused)
+        pausedTrackInfo->m_trackInfo.pausePlayedTime();        
```

Comment:
You crash is not a deadlock, it is a Segmentation fault, which happens when you access memory, not allocated by the process. In this case, you access memory via the uninitialized pointer pausedTrackInfo.

It is a good idea to always initialize empty pointers with "nullptr". Than you can decorate your code with 
DEBUG_ASSERT(pausedTrackInfo) in various places to detect such design issues early. 
If you start mixxx with --debugAssertBreak under GDB, it will hit a breakpoint, instead of just abort. 
Very handy :-)

Reply:
Wow, how can a seg fault keep all threads in a futex? Wouldn't it kill the process? Thanks for the help.

Reply:
It actually kills the process her on my Ubuntu Trusty.

Reply:
Mybe we have an additional deadlock?